Fixes: LINKER BUG: .rofixup section size mismatch

Workaround from adi-toolchain commit:
af414ec1fb628ddb1562f062f3db49823ca14ca2
[#6798] binutils-2.21/bfd: temp workaround for rofixup mismatch errors.

Signed-off-by: Waldemar Brodkorb <wbx@openadk.org>

diff -Nur binutils-2.26.1.orig/bfd/elf32-bfin.c binutils-2.26.1/bfd/elf32-bfin.c
--- binutils-2.26.1.orig/bfd/elf32-bfin.c	2015-11-13 09:27:40.000000000 +0100
+++ binutils-2.26.1/bfd/elf32-bfin.c	2016-07-31 00:50:48.357995132 +0200
@@ -4879,7 +4879,8 @@
 
 	case R_BFIN_FUNCDESC:
 	  picrel->fd++;
-	  picrel->relocsfd++;
+	  if (bfd_get_section_flags (abfd, sec) & SEC_ALLOC)
+	    picrel->relocsfd++;
 	  break;
 
         /* This relocation describes the C++ object vtable hierarchy.
